mail-server/rsnapshot.nix

```diff
@@ -0,0 +1,45 @@
+#  nixos-mailserver: a simple mail server
+#  Copyright (C) 2016-2017  Robin Raymond
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program. If not, see <http://www.gnu.org/licenses/>
+
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.mailserver;
+  preexecDefined = cfg.backup.cmdPreexec != null;
+  preexecString = optionalString preexecDefined "cmd_preexec	${cfg.backup.cmdPreexec}";
+  postexecDefined = cfg.backup.cmdPostexec != null;
+  postexecString = optionalString postexecDefined "cmd_postexec	${cfg.backup.cmdPostexec}";
+in {
+  config = mkIf cfg.backup.enable {
+    services.rsnapshot = {
+      enable = true;
+      cronIntervals = cfg.backup.cronIntervals;
+      # rsnapshot expects intervals shortest first, e.g. hourly first, then daily.
+      # tabs must separate all elements
+      extraConfig = ''
+        ${preexecString}
```

Comment:
It seems the documentation of rsnapshot requires a script to be specified here. Are you sure it works with a bash command? Otherwise we can write the command to a script and include it here.

Reply:
Sorry for the delay. I was unaware you commented on this PR.

This is correct, rsnapshot requires an absolute path to a an executable (script).

My goal is to stay true to how `rsnapshot` operates to keep things simple and easy to understand. This being said I won't lose any sleep if we wrap this in a script. So I can either do that or document it more explicitely that rsnapshot expects a path here. Thoughts @r-raymond ?